Switch to bazelmod

ratnikov opened the pull request:
Resolves https://github.com/grpc/grpc-kotlin/issues/398

I'm open to which version to pick, just happy I can depend on grpc-kotlin using bazel mod. :)

ratnikov committed:
switch to bazelmod

ratnikov committed:
Use a uniquelly named maven to avoid conflicts with other repos.

ed-irl commented:
@ratnikov I was just lurking on the issue that this patch is targeted towards. Is there an alternative/competing patch thats not listed on the issue? Nice work regardless, it would be good to see bzlmod support. Hopefully you're able to get the CLA and stuff and find a reviewer 😄 🤞

ratnikov commented:
I'm not aware of any alternative work. I've been using my patch internally for a bit, so so far it works. :)

We've been making some progress on our end to get the EasyCLA hurdle out of the way, so I'm hopping that that will be green in a week or two. I have a feeling that may elicit some input from maintainers on its mergability.

gonzojive commented:
Is anything blocking review of this? It seems the CLA has been signed.

ratnikov commented:
@lowasser hey, looks like you merged last few commits. Could you help me understand what's the best process to follow to proceed with the review of this PR?

aragos commented on MODULE.bazel.lock:
Do we need this file in the commit?

ratnikov replied:
Looks like rules_java doesn't include, so removed it and added lock file to the .gitignore.

ratnikov committed:
Remove the lockfile from check-in.

Following rules_java example seems sane to me:
https://github.com/bazelbuild/rules_java/blob/master/.gitignore#L20

gonzojive commented:
Ready to merge?

ratnikov commented:
@gonzojive Yes. Waiting for maintainers to take action and unblock running of the checks (or unblock merging in a different way).

blockjon-dd commented:
👋 bazelmod user here. I tried the techniques seen here in my project and things seem like they are working. Thanks for the great work.

cgruber commented:
@lowasser - Any movement on this? (also, hi!)

stack-plumber commented:
@ratnikov Thanks for this work. (bzlmod user here and also blaze user as I am a Googler :P) . Eagerly waiting for this PR to be merged from past few months. I can use the same tech-stack which I am currently used-to in Google for fullstack development.

dougthor42 commented:
Friendly /poke from another googler. :wave:

jack-bradshaw commented:
Is this going ahead? Bazel 9 is coming and soon workspace will not be feasible at all.
